test/jdk/java/util/Properties/StoreReproducibilityTest.java

```diff
@@ -0,0 +1,493 @@
+                    }
+                    if (!commentLine2.equals("c=d")) {
+                        throw new RuntimeException("Unexpected comment line " + commentLine2 + " in " + tmpFile);
+                    }
+                } else {
+                    if (!commentLine1.equals(storeDate)) {
+                        throw new RuntimeException("Unexpected comment line " + commentLine1 + " in " + tmpFile);
+                    }
+                }
+            }
+            assertAllFileContentsAreSame(storedFiles, storeDate);
+        }
+    }
+
+    // launches the java process and waits for it to exit. throws an exception if exit value is non-zero
+    private static void executeJavaProcess(ProcessBuilder pb) throws Exception {
+        final OutputAnalyzer outputAnalyzer = ProcessTools.executeProcess(pb);
+        try {
+            outputAnalyzer.shouldHaveExitValue(0);
+        } finally {
+            // print out any stdout/err that was generated in the launched program
+            outputAnalyzer.reportDiagnosticSummary();
+        }
+    }
+
+    // Properties.load() from the passed file and return the loaded Properties instance
+    private static Properties loadProperties(final Path file) throws IOException {
+        final Properties props = new Properties();
+        props.load(Files.newBufferedReader(file));
+        return props;
+    }
+
+    /**
+     * Verifies that the date comment in the {@code destFile} is same as {@code storeDate}
+     */
+    private static void assertExpectedStoreDate(final Path destFile,
+                                                final String storeDate) throws Exception {
+        final String dateComment = findNthComment(destFile, 2);
+        if (dateComment == null) {
+            throw new RuntimeException("Comment \"" + storeDate + "\" not found in stored properties " + destFile);
+        }
+        if (!storeDate.equals(dateComment)) {
+            throw new RuntimeException("Expected comment \"" + storeDate + "\" but found \"" + dateComment + "\" " +
+                    "in stored properties " + destFile);
+        }
+    }
+
+    /**
+     * Verifies that the date comment in the {@code destFile} can be parsed using the
+     * "EEE MMM dd HH:mm:ss zzz yyyy" format and the time represented by it is {@link Date#after(Date) after}
+     * the passed {@code date}
+     */
+    private static void assertCurrentDate(final Path destFile, final Date date) throws Exception {
+        final String dateComment = findNthComment(destFile, 2);
+        if (dateComment == null) {
+            throw new RuntimeException("Date comment not found in stored properties " + destFile);
+        }
+        System.out.println("Found date comment " + dateComment + " in file " + destFile);
+        final Date parsedDate;
+        try {
+            parsedDate = new SimpleDateFormat(dateCommentFormat).parse(dateComment);
+        } catch (ParseException pe) {
```

Comment:
Its slightly better to use the same date formatting and parsing APIs consistently.
DateTimeFormatter.parse could be used here since DateTimeFormatter was used above. 
(Though the pattern uses "yyyy" instead of "uuuu" for the year.)

Reply:
Done. I've updated the tests to use a consistent API for parsing these date comments in the stored files. They now use the DateTimeFormatter APIs along with the right pattern ("uuuu") to match the output of Date.toString().